src/model.jl

```diff
@@ -1133,16 +1538,293 @@ function update_auxvars(data::AbstractArray{Float64,2}, model::Model;
+end
+
+export findequations
+
+"""
+    equation_map(e::Model)
+    
+    Returns a dictionary with the keys beign the symbols used in the models equations 
+    and the values being a vector of equation keys for equations which use these symbols. 
+"""
+function equation_map(m::Model)
+    eqmap = Dict{Symbol, Any}()
+    for (key, eqn) in pairs(alleqns(m))
+        for (var, time) in keys(eqn.tsrefs)
+            if var.name ∈ keys(eqmap)
+                unique!(push!(eqmap[var.name], key))
+            else
+                eqmap[var.name] = [key]
+            end
+        end
+        for param in keys(eqn.eval_resid.params)
+            if param ∈ keys(eqmap)
+                unique!(push!(eqmap[param], key))
+            else
+                eqmap[param] = [key]
+            end
+        end
+        for var in keys(eqn.ssrefs)
+            if var.name ∈ keys(eqmap)
+                unique!(push!(eqmap[var.name], key))
+            else
+                eqmap[var.name] = [key]
+            end
+        end
+    end 
+    for (key, eqn) in pairs(m.sstate.constraints)
+        for ind in eqn.vinds
+            name = m.sstate.vars[(1+ind)÷2].name.name
+            if name ∈ keys(eqmap)
+                unique!(push!(eqmap[name], key))
+            else
+                eqmap[name] = [key]
+            end
+        end
+        for param in keys(eqn.eval_resid.params)
+            if param ∈ keys(eqmap)
+                unique!(push!(eqmap[param], key))
+            else
+                eqmap[param] = [key]
+            end
+        end
+    end 
+    return eqmap
+end
+
+"""
+    @replaceparameterlinks model oldmodel => newmodel
+    
+
+This function is used when a model uses parameters which link to another model object.
+The function must be called with a pair of models as they appear in the Main module.
+
+This is useful when ones models are modularized and include sattelite models. The function
+can then be used to link the parameters in modified copies of the sattelite model to modified 
+copies of the main model. For example, if the FRBUS_VAR model has a main model and a sattelite model
+the following workflow would make sense.
+
+```
+using FRBUS_VAR
+m = deepcopy(FRBUS_VAR.model)
+m_sattelite = deepcopy(FRBUS_VAR.sattelitemodel)
+
+## INSERT CHANGES to m
+@reinitialize m
+@replaceparameterlinks m_sattelite FRBUS_VAR.model => m
+@reinitialize m_sattelite
+
+```
+
+Changes like this should be followed by a call to [`@reinitialize`](@ref) on the model.
+"""
+macro replaceparameterlinks(model, expr)
+    if expr.args[1] !== :(=>)
+        error("The replacement must by of the form oldmodel => newmodel")
+    end
+    old_string = string(expr.args[2])
+    new_string = string(expr.args[3])
+    model_name = string(model)
+    return esc(:(
+        params_string = ModelBaseEcon.corrected_parameters_block($model, $old_string, $new_string, $model_name);
+        eval(Meta.parse(params_string));
+        nothing
+        ));
+end
```

Comment:
Let's fix this macro.  Instead of converting to string, doing string processing to fix the string, then parsing and evaluating it, it'd be better to work directly with symbols and expressions.

Reply:
This was difficult, it turns out the key change needed was to replace the module of the Parameters object to the `Main` module. I'm a bit concerned that one consequence is that a deepcopied model would still have its origin module as the module for parameter evaluation, whereas the deepcopied sattelite model on which we run `@replaceparameterlinks` now has the Main module as the module for parameter evaluation. 

Any thoughts you might have on this would be welcome. https://github.com/bankofcanada/ModelBaseEcon.jl/pull/53/commits/02f527e1b44d098711901ff962d3d8b5b9ef64d7

Reply:
There may also be scoping issues if @replaceparameterlinks this is run within a begin block.